File: 프로그래머스/lv2/84512. 모음 사전/모음 사전.swift
import Foundation

let arr = ["A","E","I","O","U"]
var ans = [String]()

func solution(_ word:String) -> Int {
    var a = 0
    for i in 1...5{
        dfs(i,"")
    }
    ans.sort()
    for i in 0...ans.count-1{
        if(ans[i] == word){
            a = i + 1
            break
        }
    }
    return a
}

func dfs(_ n:Int,_ str:String){
    var str = str
    if(str.count < n){
        for i in arr{
            str += i
            dfs(n,str)
            str.removeLast()
        }
    }
    else{
        ans.append(str)
        return
    }
}

File: 프로그래머스/2/84512. 모음 사전/모음 사전.java
import java.util.*;

class Solution {
        String [] arr = {"A","E","I","O","U"};
        ArrayList arrayList = new ArrayList<>();
        public int solution(String word) {
            int answer = 0;
           
            fun("");
            Collections.sort(arrayList);
            for (int i = 0; i < arrayList.size(); i++) {
                System.out.println(arrayList.get(i)+" "+word);
                if (arrayList.get(i).equals(word)){
                    answer = i+1;
                    break;
                }
            }
            return answer;
        }

        public void fun(String word){
            String a = word;
            if (word.length() == 5){
                return;
            }
            for (int i = 0; i < arr.length; i++) {
                String b = a+arr[i];
                arrayList.add(b);
                fun(b);
            }
        }
    }
